src/opensles/AudioStreamOpenSLES.cpp

```diff
@@ -123,35 +123,61 @@ SLresult AudioStreamOpenSLES::finishCommonOpen(SLAndroidConfigurationItf configI
     return SL_RESULT_SUCCESS;
 }
 
-Result AudioStreamOpenSLES::configureBufferSizes(int32_t sampleRate) {
-    LOGD("AudioStreamOpenSLES:%s(%d) initial mFramesPerBurst = %d, mFramesPerCallback = %d",
-            __func__, sampleRate, mFramesPerBurst, mFramesPerCallback);
-    // Decide frames per burst based on hints from caller.
-    if (mFramesPerCallback != kUnspecified) {
-        // Requested framesPerCallback must be honored.
-        mFramesPerBurst = mFramesPerCallback;
-    } else {
-        mFramesPerBurst = DefaultStreamValues::FramesPerBurst;
-
-        // Calculate the size of a fixed duration high latency buffer based on sample rate.
-        int32_t framesPerHighLatencyBuffer =
-                (kHighLatencyBufferSizeMillis * sampleRate) / kMillisPerSecond;
+int32_t AudioStreamOpenSLES::calculateOptimalBufferQueueLength() {
+    int32_t queueLength = kBufferQueueLengthDefault;
+    int32_t likelyFramesPerBurst = estimateNativeFramesPerBurst();
+    // What is required based on capacity?
+    if (mBufferCapacityInFrames > 0) {
+        int32_t queueLengthFromCapacity = (mBufferCapacityInFrames + likelyFramesPerBurst - 1)
+                / likelyFramesPerBurst;
+        queueLength = std::max(queueLength, queueLengthFromCapacity);
+    }
+    // What is required based on requested callback size?
+    if (mFramesPerCallback > 0) {
+        int32_t minCapacity = 2 * mFramesPerCallback;
```

Comment:
Should we use a constant for this 2? Or at least a comment on double buffering?

Reply:
OK